draft-birkholz-scitt-receipts.md

```diff
@@ -62,22 +78,30 @@ This document defines a method for countersigning of COSE_Sign1 messages when us
 
 # Common Parameters
 
-A Transparency Service backed by a Merkle tree is defined by a collection of immutable parameters, which are used by clients to communicate with the service and to verify artifacts. Each of these parameters must be established before the service operator begins to operate the service. At minimum, the following must be defined:
+Verifiers are configured by a collection of parameters 
+to identify a Transparency Service and verify its Receipts. 
+These parameters MUST be fixed for the lifetime of the Transparency Service
+and securely communicated to all Verifiers. 
+
+At minimum, these parameters include:
 
-- Service identifier: An opaque identifier (e.g. UUID) that uniquely identifies the service and all other parameters.
+- a Service identifier: An opaque identifier (e.g. UUID) that uniquely identifies the service and all other parameters.
 
-- Tree algorithm: The tree algorithm used. This document creates a registry (see {{tree-alg-registry}}) with an initial set of tree algorithms that are described in this document.
+  > is it sufficient? why putting it in every receipt then? What are those other parameters? I was expecting a public key or a certificate. 
 
-Additional parameters may be defined depending on the tree algorithm used.
+- The Tree algorithm used for issuing receipts, and their additional global parameters, if any. This document creates a registry (see {{tree-alg-registry}}) and describes an initial set of tree algorithms.
 
-# Basic Receipt Structure
+  > The architecture also has fixed TS registration policies. 
 
-A receipt represents a countersignature produced by a Transparency Service.
-The receipt structure is a CBOR array. The fields of the array in order are:
+# Generic Receipt Structure
 
-- service_id: The service identifier as tstr.
+A Receipt represents a countersignature issued by a Transparency Service.
 
-- contents: The receipt contents as a CBOR structure determined by the tree algorithm.
+The Receipt structure is a CBOR map with two items, in order: 
```

Comment:
Consider CBOR arrays with fixed structure as C-style structs. If there is no need for extensibility then using a map is typically not done in CBOR (e.g. the top-level COSE structure is defined as such an array). If we do use maps, we have to decide on the data type of the keys (int vs string). @henkbirkholz for additional input.

Reply:
Ok thanks. Just to confirm: this means the field labels are omitted from the wire format.